plugins/jobs/src/js/types/JobConnection.ts

```diff
@@ -1,19 +1,107 @@
-import { Job } from "#PLUGINS/jobs/src/js/types/Job";
+import { Job, JobTypeResolver } from "#PLUGINS/jobs/src/js/types/Job";
+import { JobResponse as MetronomeJobResponse } from "src/js/events/MetronomeClient";
 
+import JobStates from "#PLUGINS/jobs/src/js/constants/JobStates";
+import JobStatus from "#PLUGINS/jobs/src/js/constants/JobStatus";
 export interface JobConnection {
+  path: string[];
   filteredCount: number;
   totalCount: number;
   nodes: Job[];
 }
 
 export const JobConnectionSchema = `
 type JobConnection {
+  path: [String]!
   filteredCount: Int!
   totalCount: Int!
   nodes: [Job]!
 }
 `;
 
-export function JobConnectionTypeResolver(response: any): any {
-  return response;
+export interface JobsQueryArgs {
+  path: string[];
+  filter?: string;
+  sortBy?: SortOption;
+  sortDirection?: SortDirection;
+}
+
+export type SortOption = "ID" | "STATUS" | "LAST_RUN";
+
+export type SortDirection = "ASC" | "DESC";
+
+export function JobConnectionTypeResolver(
+  response: MetronomeJobResponse[],
+  args: JobsQueryArgs
+): JobConnection {
+  const { sortBy = "ID", sortDirection = "ASC", filter = "", path = [] } = args;
+
+  const namespacedResponse: MetronomeJobResponse[] =
+    path.length > 0
+      ? response.filter(({ id }) => id.startsWith(path.join(".")))
+      : response;
+
+  const filteredResponse: MetronomeJobResponse[] =
+    filter !== ""
+      ? namespacedResponse.filter(({ id }) => id.includes(filter))
+      : namespacedResponse;
+
+  const resolvedJobs = filteredResponse.map(JobTypeResolver);
+
+  const sortedJobs: Job[] = resolvedJobs.sort((a, b) =>
+    compareJobs(a, b, sortBy, sortDirection)
+  );
+
+  // I decided not to go with fieldresolvers here, because the logic is a bit more complex and we need to find a better solution first
```

Comment:
Can you explain why you choose to not use field resolvers? I understand that the sorting and filtering prevents you from using them, but we could do this work within a function and reuse it. Or we could make the field resolvers work on the filtered responses.

I would really like us to stick in the pattern of field resolvers, but if you think that is more than 30 minutes of work, let's sync on it

Reply:
I agree in general, but in this particular case I decided to keep it all in one because of the potential performance impact of calling the whole thing twice (for filteredCount & nodes). also we already know that we need to redo this… 🤷‍♂️

Reply:
Then let's have a ticket for refactoring this to field resolvers